### PR TITLE
Add MNIST torchvision example

### PR DESCRIPTION
**Pull request by ottonemo**
Utilizes and tests `skorch.helper.SliceDataset` to do grid search and
plotting of examples.

### CODE REVIEW

**Comment by ottonemo** [comment]
@thomasjpfan @BenjaminBossan revised, please review again.

**Comment by BenjaminBossan** [comment]
LGTM

**Comment by thomasjpfan** [comment]
There is a new `skorch/tests/net_cuda.pkl` file in this PR.

File: notebooks/MNIST-torchvision.ipynb [hunk 1]
```diff
@@ -0,0 +1,847 @@
...
+    "If you are running in colab, you should install the dependencies and download the dataset by running the following cell:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! [ ! -z \"$COLAB_GPU\" ] && pip install torch scikit-learn==0.20.* skorch"
```
[the remark this answers is not in the document]
Comment (ottonemo) [reply notebooks/MNIST-torchvision.ipynb]: No, I reran the example with sklearn 0.21 and updated the cell.

**Comment by ottonemo** [comment]
@thomasjpfan I reset `net_cuda.pkl` to master, however we should probably update the pickle whenever there is a change (to keep the CI tests that depend on this pickle up-to-date). It's just not in the scope of this PR.

Please review again.

**Comment by ottonemo** [comment]
@thomasjpfan is this good to go?

**Pull request merged**